### ISSUE TITLE
Allow env/env-file to be used alongside system values

### ISSUE THREAD

**Issue by ajcrites** [danday74/envsub#9]
Currently if you use `--env` or `--env-file`, there is no way to also include system environment variables. For example:

```
# .env
ENV1=env1
ENV2=env2
```

```
# this will only substitute ENV3
$ export ENV3=env3 envsub file1 file2
```

```
# this will only substitute ENV1 and ENV2
$ export ENV3=env3 envsub file1 file2 --env-file .env
```

I think there are two possible ways to handle this:

1. allow `--system` to work with `--env-file` like it works with `--env`
2. provide another flag like `--env-system` that works similar to `--env-file` except it uses the system environment as well.

**Comment by ajcrites** [comment]
I notice that you're busy, so I'd be happy to make this change. I'm just interested to know if there is one you would prefer.

**Comment by danday74** [comment]
Given the increase in downloads on this module I am planning to be more active - will investigate this issue and get back to you

**Comment by danday74** [comment]
Research on this issue:

```envsub templateFile outputFile --env VARFROMSYSTEM```
means that only VARFROMSYSTEM will be substituted

**Comment by danday74** [comment]
Having reviewed this I agree with @ajcrites statement:

> if you use --env or --env-file, there is no way to also include system environment variables

I also agree this needs fixing. With regards to proposed solutions:

1) allow --system to work with --env-file like it works with --env
The `--system` flag already works with the `--env-file` flag, preferring to substitute with system values where they exist as opposed to default values provided in the env file. This is not a suitable solution.

2) provide another flag like --env-system that works similar to --env-file except it uses the system environment as well.

This sounds like a great solution. This should not modify the existing behaviour of `--env` and `--env-file` flags but, once those substitutions have been done, should also substitute system environment variables - ensuring not to modify the values of substitutions that have already taken place. As such, this flag should be considered lower priority than the `--env` and `--env-file` flags.

**Comment by danday74** [comment]
Ended up introducing the `--all` flag which does what you want - thanks :)
will be released in 4.0.3
